build_runner/lib/src/logging/std_io_logging.dart

```diff
@@ -5,27 +5,22 @@
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:io/ansi.dart';
 import 'package:logging/logging.dart';
 import 'package:stack_trace/stack_trace.dart';
 
-final _cyan = _isPosixTerminal ? '\u001b[36m' : '';
-final _yellow = _isPosixTerminal ? '\u001b[33m' : '';
-final _red = _isPosixTerminal ? '\u001b[31m' : '';
-final _endColor = _isPosixTerminal ? '\u001b[0m' : '';
-final _isPosixTerminal =
-    !Platform.isWindows && stdioType(stdout) == StdioType.TERMINAL;
-
 void stdIOLogListener(LogRecord record) {
-  String color;
+  AnsiCode color;
   if (record.level < Level.WARNING) {
-    color = _cyan;
+    color = cyan;
   } else if (record.level < Level.SEVERE) {
-    color = _yellow;
+    color = yellow;
   } else {
-    color = _red;
+    color = red;
   }
-  var header = '${_isPosixTerminal ? '\x1b[2K\r' : ''}'
-      '$color[${record.level}]$_endColor ${record.loggerName}: '
+  final level = color.wrap('[${record.level}]');
+  var header = '${ansiOutputEnabled ? '\x1b[2K\r' : ''}'
```

Comment:
Note that I think `ansiOutputEnabled` is actually using different logic than `_isPosixTerminal` was but it probably doesn't matter (I think it might be more correct anyways).

Reply:
Yeah it's definitely different. I'm not sure it's more correct (it failed on my terminal where the old logic succeeded https://github.com/dart-lang/sdk/issues/31606) but it's at least more standard in Dart...